avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -935,7 +939,20 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
           int precision = precisionString == null ? CONNECT_AVRO_DECIMAL_PRECISION_DEFAULT :
               Integer.parseInt(precisionString);
           int scale = scaleString == null ? 0 : Integer.parseInt(scaleString);
-          org.apache.avro.LogicalTypes.decimal(precision, scale).addToSchema(baseSchema);
+          if (scale < 0 || scale > precision) {
+            log.debug(
+                "Scale and precision of {} and {} cannot be serialized as native Avro logical " 
+                    + "decimal type; reverting to legacy serialization method",
+                scale,
+                precision
+            );
```

Comment:
Do we really want this as DEBUG?

Reply:
I'm afraid to put it much higher as this will be invoked potentially multiple times on every record and would result in log spam. Could put it at `TRACE` if you feel that `DEBUG` is too high, though.

Reply:
I was thinking DEBUG is too high, considering this might be output for potentially every record (or at least a lot of records) from a source connector. Might be useful to have a DEBUG message that happens once per connector instance, but more than that seems too many.

Reply:
I can put it down to `TRACE` for now... I agree that it may be useful to add a once-per-connector (or, more accurately, once-per converter instance) message at `DEBUG` level but tracking that kind of state in this class would come with its own implementation complexity and complicate an already several-thousand-line-long file. I think that, given that all we're doing here is reverting to prior behavior in a case where things used to work and now break, it should be safe to put at `TRACE` for now.